### PR TITLE
Add preliminary support for subscription keychain sharing

### PR DESCRIPTION
**Pull request by quanganhdo**
Task/Issue URL: https://app.asana.com/0/414235014887631/1206564116141608/f
Tech Design URL:
CC:

**Description**:

This PR does the followings:

- Adding a new subscriptions app group
- Injecting this app group into the subscriptions library to be used to create keychain entries.

**Steps to test this PR**:

It shouldn’t break any existing flow.

**Copy Testing**:

* [ ] Use of correct apostrophes in new copy, ie `’` rather than `'`

**Orientation Testing**:

* [ ] Portrait
* [ ] Landscape

**Device Testing**:

* [ ] iPhone SE (1st Gen)
* [ ] iPhone 8
* [ ] iPhone X
* [ ] iPhone 14 Pro
* [ ] iPad

**OS Testing**:

* [ ] iOS 14
* [ ] iOS 15
* [ ] iOS 16

**Theme Testing**:

* [ ] Light theme
* [ ] Dark theme

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

### CODE REVIEW

File: Core/BundleExtensions.swift [hunk 1]
```diff
@@ -0,0 +1,40 @@
+//
+//  BundleExtensions.swift
```
Comment (miasma13) [inline Core/BundleExtensions.swift]: A nitpick but I'd go for singular `BundleExtension`

**Review by miasma13 — COMMENTED** [review]
WDYT about using a pattern that @graeme used for macOS and from the providing a convenience initializer to the changes with the `AccountManager` on most places?

See https://github.com/duckduckgo/macos-browser/blob/6231a8e6c456c0f35a1498e2883ecc0d55d93b48/DuckDuckGo/Subscription/AccountManagerExtension.swift

File: DuckDuckGo/Subscription/UserScripts/SubscriptionPagesUseSubscriptionFeature.swift [hunk 2]
```diff
@@ -319,7 +322,7 @@ final class SubscriptionPagesUseSubscriptionFeature: Subfeature, ObservableObjec
     func restoreAccountFromAppStorePurchase() async throws {
         setTransactionStatus(.restoring)
         
-        let result = await AppStoreRestoreFlow.restoreAccountFromPastPurchase()
+        let result = await AppStoreRestoreFlow.restoreAccountFromPastPurchase(appGroup: Bundle.main.appGroup(bundle: .subs))
```
Comment (quanganhdo) [inline DuckDuckGo/Subscription/UserScripts/SubscriptionPagesUseSubscriptionFeature.swift]: I think the param name should be `subscriptionAppGroup` for consistency @miasma13.

**Review by quanganhdo — COMMENTED** [review]
Updated to use convenience init.

File: DuckDuckGo/SettingsViewModel.swift [hunk 3]
```diff
@@ -347,7 +347,7 @@ extension SettingsViewModel {
         
         // Fetch available subscriptions from the backend (or sign out)
         switch await SubscriptionService.getSubscriptionDetails(token: token) {
-        
+
```
Comment (miasma13) [inline DuckDuckGo/SettingsViewModel.swift]: This will make linter cry

**Review by miasma13 — APPROVED** [review]
LGTM 👍

**Pull request merged**